File: buy-sell-stock/answer.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0:
            return 0
        
        minPrice = 100000
        maxProfit = 0
        
        for price in prices:
            if price < minPrice:
                minPrice = price
            elif price - minPrice > maxProfit:
                maxProfit = price - minPrice
        
        return maxProfit

File: buy-sell-stock/answer.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max = 0;
        int min = *max_element(prices.begin(), prices.end());
        
        for (int i = 0; i < prices.size(); i++) {
            if (prices[i] < min) {
                min = prices[i];
            } else {
                max = max > prices[i] - min ? max : prices[i] - min;
            }
        }
        
        return max;
    }
};
